src/Accounts/Accounts/Context/GetAzureRMContext.cs

```diff
@@ -84,6 +84,7 @@ public override void ExecuteCmdlet()
                     var defaultProfile = DefaultProfile as AzureRmProfile;
                     if (defaultProfile != null && string.Equals(AzureSession.Instance?.ARMContextSaveMode, "CurrentUser"))
                     {
+                        AzureSession.Instance.SetProperty(AzureSession.Property.Environment, DefaultContext.Environment.Name);
```

Comment:
Why need to update session in a Get cmdlet which usually doesn't have side effects?

Reply:
> Why need to update session in a Get cmdlet which usually doesn't have side effects?

The value is used in https://github.com/Azure/azure-powershell/blob/a39b7bfd2c824ed24aeb741bf62d5f1bf7e871c0/src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs#L822 which is used in `Get-AzContext -ListAvailable -RefreshContextFromTokenCache`

Reply:
I just feel it strange to set the environment when getting the context. Shouldn't it be set when during connect-azaccount?

Reply:
> I just feel it strange to set the environment when getting the context. Shouldn't it be set when during connect-azaccount?

It is already set in Connect-AzAccount. But when the customers restart the powershell and run Get-AzContext -ListAvailable -RefreshContextFromTokenCache without login, the contexts are going to be cleared. That's why we have to set it here.